internal/codespaces/api/api.go

```diff
@@ -608,15 +609,16 @@ func (a *API) CreateCodespace(ctx context.Context, params *CreateCodespaceParams
 }
 
 type startCreateRequest struct {
-	RepositoryID       int    `json:"repository_id"`
-	IdleTimeoutMinutes int    `json:"idle_timeout_minutes,omitempty"`
-	Ref                string `json:"ref"`
-	Location           string `json:"location"`
-	Machine            string `json:"machine"`
-	DevContainerPath   string `json:"devcontainer_path,omitempty"`
-	VSCSTarget         string `json:"vscs_target,omitempty"`
-	VSCSTargetURL      string `json:"vscs_target_url,omitempty"`
-	PermissionsOptOut  bool   `json:"multi_repo_permissions_opt_out"`
+	RepositoryID           int    `json:"repository_id"`
+	IdleTimeoutMinutes     int    `json:"idle_timeout_minutes,omitempty"`
+	RetentionPeriodMinutes int    `json:"retention_period_minutes,omitempty"`
```

Comment:
@reybard pointed out that this won't serialize a 0-minute retention period with `omitempty` so I'll add a test and fix that

Reply:
Just to confirm: the user could specify `0` here, in order to indicate they want to delete it immediately upon idle?

Reply:
@lostintangent That is correct. It may not be immediate as it's a scheduled job but it should be pretty quick

Reply:
Awesome, thanks for confirming that!